sw/device/lib/base/memory.c

```diff
@@ -60,14 +170,40 @@ int OT_PREFIX_IF_NOT_RV32(memcmp)(const void *lhs, const void *rhs,
 OT_WEAK
 int OT_PREFIX_IF_NOT_RV32(memrcmp)(const void *lhs, const void *rhs,
                                    size_t len) {
-  const uint8_t *lhs8 = (uint8_t *)lhs;
-  const uint8_t *rhs8 = (uint8_t *)rhs;
-  size_t j;
-  for (size_t i = 0; i < len; ++i) {
-    j = len - 1 - i;
-    if (lhs8[j] < rhs8[j]) {
+  if (len == 0) {
+    return kMemCmpEq;
+  }
+  const unsigned char *lhs8 = (const unsigned char *)lhs;
+  const unsigned char *rhs8 = (const unsigned char *)rhs;
+  size_t leading_bytes_end, aligned_words_end;
+  compute_alignment((uintptr_t)lhs, (uintptr_t)rhs, len, &leading_bytes_end,
+                    &aligned_words_end);
+  size_t end = len;
```

Comment:
nit: `end` is initialized to end but not always the end. what about 
```
size_t i = len - 1;
for (; i >= aligned_words_end; --i) {
...
}
```

There are probably more than one off-by-one errors in my example hehe

Reply:
Ah, I intentionally avoided that phrasing to avoid the tautology of `for (; i >= 0; --i)`. As it is now, I can remove the `if (len == 0)` check at the beginning, but if we replaced `end` with `last_index` or something, I think we'd need to bring the length zero-check back. WDYT?

Reply:
We can drop this, thanks!